llama_index/indices/keyword_table/retrievers.py

```diff
@@ -87,6 +87,9 @@ def _retrieve(
         chunk_indices_count: Dict[str, int] = defaultdict(int)
         keywords = [k for k in keywords if k in self._index_struct.keywords]
         logger.info(f"> Extracted keywords: {keywords}")
+        if (len(keywords) == 0):
+            node_id = self._index_struct.table[self._index_struct.keywords.pop()]
+            chunk_indices_count[node_id.pop()] += 1
```

Comment:
@aidenchen01 Thanks for the contribution! Could you explain the proposed change in your PR summary? It would really help us give a good review and get this in!

Reply:
sure, the change for this part is in some uncommon situation, where your query has no relationship to your docs and the Index will found no keywords matched with your query, and chunk_indices_count variable will remain null  causing error in subsequent execution like the case I show above

Reply:
yep, question is more about the proposed logic for fixing this. I'm just trying to figure out if/why this is the right fix?

Reply:
To ensure at least one keyword matched we take the first key word in the index  as default. May be not the best choice, but a quick temp fix. I think it's not a ultimate fix for this, but as long as the ultra fix come along, people will run into this error eventually, so quick fix is needed

Reply:
I also realized the former choice of pop to get element in a set is not proper, so I changed the code getting element from it

Reply:
Personally, I think a better fix is returning an empty response (instead of hitting the error about zero division). 

If no keywords match, I don't think it makes sense to use a random keyword 🤔

Reply:
@aidenchen01 i agree with @logan-markewich on this one. is it possible to make this change?